bigquery/params.go

```diff
@@ -129,12 +139,57 @@ type QueryParameter struct {
 	Value interface{}
 }
 
-func (p QueryParameter) toBQ() (*bq.QueryParameter, error) {
+// QueryParameterValue is a go type for representing a explicit typed BigQuery Query Parameter.
```

Comment:
I'm still on the fence about the name due to the collision with the API, but it's a reasonable name otherwise.  

Should we document the behavior for nested types as part of the type description?  Specifically that you need to fully specify types and values for complex params?